ml-agents/mlagents/trainers/torch/networks.py

```diff
@@ -82,20 +108,49 @@ def forward(
         sequence_length: int = 1,
     ) -> Tuple[torch.Tensor, torch.Tensor]:
         encodes = []
+        var_len_inputs = []  # The list of variable length inputs
+        var_len_processors = [
+            p for p in self.processors if isinstance(p, EntityEmbedding)
+        ]
+
         for idx, processor in enumerate(self.processors):
-            obs_input = inputs[idx]
-            processed_obs = processor(obs_input)
-            encodes.append(processed_obs)
+            if not isinstance(processor, EntityEmbedding):
+                # The input can be encoded without having to process other inputs
+                obs_input = inputs[idx]
+                processed_obs = processor(obs_input)
+                encodes.append(processed_obs)
+            else:
+                var_len_inputs.append(inputs[idx])
```

Comment:
I like this better, at least for this part of the code. Can't to speak to whether making the "add_self_embedding" change is worth it.

One other thing that would (IMHO) make this section cleaner is to store`(processor, inputs[idx])` here instead; that way you don't need to form `var_len_processors`, and don't need the `zip()`

Reply:
Done